samples/AppConfiguration/Generated/AppConfigurationRestClient.cs

```diff
@@ -92,7 +93,7 @@ internal HttpMessage CreateGetKeysRequest(string name, string after, string acce
                         return ResponseWithHeaders.FromValue(value, headers, message.Response);
                     }
                 default:
-                    throw await ClientDiagnostics.CreateRequestFailedExceptionAsync(message.Response).ConfigureAwait(false);
```

Comment:
Curious: could there be any issues with sync/async here?  Are we not worried about that because we expect a buffered stream to only hold an error response and so be small?

Reply:
Yep, that's right - we are potentially paying a price by using sync to buffer if the response is not buffered (only applies to a handful of libraries), but I believe our plan was to always migrate to using RFE ctor.

Reply:
Good point - this should probably switch on sync/async

Reply:
But since it is not a ctor, there is no async option, I guess.

Reply:
Yeah, and we are buffering synchronously anyway due to the new Response property.

Reply:
> It should be mitigated by the fact that this is an error scenario, and we expect error response bodies to be small.

Yeah, I think that is probably a safe assumption.  I guess for completeness, we could put in a size check and fail buffering in the constructor if the stream starts to get to big ... what do you think?  It feels like a risk mitigation tradeoff vs. implementation complexity.  What would worst case scenario be if the error stream was large for some reason?

Reply:
This would need to happen in https://github.com/Azure/azure-sdk-for-net/pull/35716 if we added it. I would vote to not add it, as this would be no worse than the sync case of buffering the response, and we would also have to think about whether the limit should apply only for async paths.

Reply:
Yep, I agree it would be implemented in the other PR.  And I would be fine with whatever outcome we choose -- I guess I'm wondering if we can predict what the outcome would be if for some reason a service sent back a large payload in an error response for some reason and we were buffering it synchronously.  Would the app using our client throw an OutOfMemoryException at some point?  Is there the possibility it would hang?  Etc?  Just so we understand the implications reasonably well.

Reply:
The experience should be no different than a customer would have now who is using sync code. We aren't adding sync over async or anything like that.

Reply:
What is the experience in sync code?

Reply:
Sync applications are generally less responsive as they have to block on IO rather than freeing up the thread for other work. So in theory, if an application is having a bunch of these exceptions simultaneously and for some reason the response content is very large, it could cause a bottleneck.

Reply:
Ok, I see.  So, in that case, an async app could start to behave like a sync app, but only if all the concurrent tasks/threads got these exceptions simultaneously (which seems like it would be pretty rare).  It's not that it could lock up the one main/UI thread and hang for some long period of time unexplainedly?

Reply:
Right, that is my thinking.

Reply:
Cool.  Yeah, then if it would require a truly exceptional case to cause something really bad to happen in a customer app, I'm fine with not adding a check.